src/test/ui/span/macro-span-replacement.stderr

```diff
@@ -2,11 +2,14 @@ warning: struct is never constructed: `S`
   --> $DIR/macro-span-replacement.rs:7:14
    |
 LL |         $b $a;
-   |              ^
+   |            --^
+   |            |
+   |            help: if this is intentional, prefix it with an underscore: `_S`
```

Comment:
NOTE: This seems a regression to me. I don't think it's a blocker but I'll file a new issue later.

Reply:
The fix would likely just require to check the span data, or, if that doesn't work (because the span isn't annotated as being part of the macro) with a `span_to_snippet` check. The later is slower, but we're already emitting an error, so that's fine.

Reply:
Is it feasible to tag the occurrence of `S` in the macro use? Perhaps by walking the expansion stack? Or am I just having a waking dream here?